taca_ngi_pipeline/deliver/deliver_grus.py

```diff
@@ -463,28 +462,31 @@ def _set_pi_details(self, given_pi_email=None):
             logger.error("Cannot fetch PI SNIC id using snic API. Error says: {}".format(str(e)))
             raise e
     
-    def _set_binfo_details(self):
+    def _set_other_member_details(self, other_member_emails=[]):
         """
-            Set bioinfo contact details if avilable, this is not mandatory so
+            Set other contact details if avilable, this is not mandatory so
             the method will not raise error if it could not find any contact
         """
-        self.binfo_email, self.binfo_snic_id = (None, None)
-        # try getting bioinfo contact email
+        self.other_member_snic_ids = []
+        # try getting appropriate contact emails
         try:
-            self.binfo_email = self._get_order_detail()['project_bx_email']
-            assert self.binfo_email #check its not empty string
-            logger.info("Bioinfo contact for project {} found: {}".format(self.projectid, self.binfo_email))
+            prj_order = self._get_order_detail()
+            owner_email = prj_order.get('owner', {}).get('email')
+            if owner_email and owner_email != self.pi_email and owner_email not in other_member_emails:
+                other_member_emails.append(owner_email)
+            binfo_email = prj_order.get('fields', {}).get('project_bx_email')
+            if binfo_email and binfo_email != self.pi_email and binfo_email not in other_member_emails:
+                other_member_emails.append(binfo_email)
         except:
-            logger.warning("Was not able to get bioinfo contact for project {}".format(self.projectid))
-        # try getting snic id for bioinfo contact
-        if self.binfo_email == self.pi_email:
-            logger.info("Bioinfo contact is same as PI, so will be ignored and not be used")
-        elif self.binfo_email:
+            pass # nothing to worry, just move on
```

Comment:
do you expect an exception here? if so, perhaps you could be specific about which type of exception to catch. if you don't expect any exceptions, you should probably skip the try-except construct altogether